src/Sarif.WorkItems/SarifWorkItemModel.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.WorkItems;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public class SarifWorkItemModel : WorkItemModel<SarifWorkItemContext>
+    {
+        public void InitializeFromLog(SarifLog sarifLog, SarifWorkItemContext context = null)
```

Comment:
>InitializeFromLog [](start = 20, length = 17)

Consider making this a static method so you can write
```
var model = SarifWorkItemModel.CreateFromLog(sarifLog);
```
instead of
```
var model = new SarifWorkItemModel();
model.InitializeFromLog(sarifLog);
```

Reply:
Went for a ctor for now. described issues w.that approach in another comment. we can utilize a static method (perhaps a TryCreate) when future circumstances warrant.

---
In reply to: [381022979](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381022979) [](ancestors = 381022979)